### DISCUSSION TITLE
dat and idx offsets disagree

### DISCUSSION THREAD

**Discussion by ericlingit** [seaweedfs/seaweedfs#4285] [Q&A] [answered]
From reading Facebook's [Haystack paper summary](https://engineering.fb.com/2009/04/30/core-data/needle-in-a-haystack-efficient-storage-of-billions-of-photos/), my understanding is that the `.idx` file stores the offsets and sizes of every needle in haystack storage (the volume file) so you don't have scan the giant haystack file to find each and every needle's offset.

With that in mind, I expect the offset values in both `.dat` and `.idx` to be the same. So, I used [`see_dat`](https://github.com/seaweedfs/seaweedfs/blob/master/unmaintained/see_dat/see_dat.go) and [`see_idx`](https://github.com/seaweedfs/seaweedfs/blob/master/unmaintained/see_idx/see_idx.go) to take a peak at the `.dat` and `.idx` files.

However, the offset values from outputs of `see_dat` and `see_idx` differ:

```
./see-dat -collection mycollection -dir . -volumeId 3
readSuperBlock volume 3 version 3
3,1b64f1b6d offset      8 size 540885(528.21 KiB) cookie b64f1b6d
3,537a07451 offset 540928 size 184767(180.44 KiB) cookie 37a07451
3,c32975ebd offset 725728 size     86(86 B)       cookie 32975ebd
```

```
./see-idx -collection mycollection -dir . -volumeId 3
key:1 offset:    1 size:540885(528.21 KiB)
key:5 offset:67616 size:184767(180.44 KiB)
key:c offset:90716 size:86    (86 B)
```

Why are their offset values different?

**Comment by ericlingit** [comment, marked as answer]
They differ by a factor of 8:
```
key 1 offset      8 / 8 = 1 
key 5 offset 540928 / 8 = 67616
key c offset 725728 / 8 = 90716
```

**Reply by kmlebedev** [reply]
It seems this is a kind of compression of idx files

**Reply by chrislusf** [reply]
this is by design. @ericlingit you can send a PR to adjust the output to only show the offset after multiplying 8.
